### PR TITLE
Added data compression support for HTTP exporter

### PR DESCRIPTION
**Pull request by mamunto**
HTTP client didn't have any compression support. In this PR I made following changes 
* Extracted `DataCompression` from DataDogExporter and made new target 
* Introduced `gzip` and `deflate`  compression support

### CODE REVIEW

**Comment by mamunto** [comment]
@nachoBonafonte any thought on Linux platform support? We will need a completely different cross-platform library. Let's discuss to find a solution.

File: Sources/Exporters/DataCompression/DataCompression.swift [hunk 1]
```diff
@@ -28,9 +28,11 @@
 
 
 import Foundation
+
+#if canImport(Compression)
 import Compression
```
Comment (mamunto) [inline Sources/Exporters/DataCompression/DataCompression.swift:33]: Looks like macOS and Linus both fail with that solution. @nachoBonafonte @bryce-b let me know if there is anywhere else I need to address the platform issue. Most probably in the `Package.swift` file but not entirely sure where

File: Sources/Exporters/OpenTelemetryProtocolHttp/OtlpHttpExporterBase.swift [hunk 2]
```diff
@@ -28,20 +31,44 @@ public class OtlpHttpExporterBase {
...
+            case .deflate:
+                if let data = rawData.deflate() {
+                    compressedData = data
+                    request.setValue("deflate", forHTTPHeaderField: "Content-Encoding")
+                }
+                
+            case .none:
+                break
+            }
+#endif
```
Comment (mamunto) [inline Sources/Exporters/OpenTelemetryProtocolHttp/OtlpHttpExporterBase.swift:62]: Did same

File: Sources/Exporters/DataCompression/DataCompression.swift (hunk 1 above)
Comment (bryce-b) [reply Sources/Exporters/DataCompression/DataCompression.swift:33]: I think the issue is you're importing DataCompression in the exporters when there's nothing there due to the #if

Comment (mamunto) [reply Sources/Exporters/DataCompression/DataCompression.swift:33]: Looks Test was a blocker. Fixed it. Thanks for the tip!

**Comment by mamunto** [comment]
It would be nice if we can include this PR and [here is another](https://github.com/open-telemetry/opentelemetry-swift/pull/634) in next release. Really appreciate your time and help

**Review by nachoBonafonte — APPROVED** [review]

**Review by vvydier — APPROVED** [review]
looks good to me. All the tests passed as well

**Pull request merged**